l2-contracts/contracts/ConsensusRegistry.sol

```diff
@@ -0,0 +1,444 @@
+        node.validatorLatest.active = false;
+
+        emit NodeDeactivated(_nodeOwner);
+    }
+
+    /// @notice Activates a previously inactive node, allowing it to participate in committees.
+    /// @dev Only callable by the contract owner or the node owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner to be activated.
+    function activate(address _nodeOwner) external onlyOwnerOrNodeOwner(_nodeOwner) {
+        _verifyNodeOwnerExists(_nodeOwner);
+        Node storage node = nodes[_nodeOwner];
+        if (_finalizeNodeRemoval(_nodeOwner, node)) {
+            return;
+        }
+
+        _ensureAttesterSnapshot(node);
+        node.attesterLatest.active = true;
+        _ensureValidatorSnapshot(node);
+        node.validatorLatest.active = true;
+
+        emit NodeActivated(_nodeOwner);
+    }
+
+    /// @notice Removes a node from the registry.
+    /// @dev Only callable by the contract owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner to be removed.
+    function remove(address _nodeOwner) external onlyOwner {
+        _verifyNodeOwnerExists(_nodeOwner);
+        Node storage node = nodes[_nodeOwner];
+
+        _ensureAttesterSnapshot(node);
+        node.attesterLatest.pendingRemoval = true;
+        _ensureValidatorSnapshot(node);
+        node.validatorLatest.pendingRemoval = true;
+
+        emit NodeRemoved(_nodeOwner);
+    }
+
+    /// @notice Changes the validator weight of a node in the registry.
+    /// @dev Only callable by the contract owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner whose validator weight will be changed.
+    /// @param _weight The new validator weight to assign to the node.
+    function changeValidatorWeight(address _nodeOwner, uint32 _weight) external onlyOwner {
+        _verifyNodeOwnerExists(_nodeOwner);
+        Node storage node = nodes[_nodeOwner];
+        if (_finalizeNodeRemoval(_nodeOwner, node)) {
+            return;
+        }
+
+        _ensureValidatorSnapshot(node);
+        node.validatorLatest.weight = _weight;
+
+        emit NodeValidatorWeightChanged(_nodeOwner, _weight);
+    }
+
+    /// @notice Changes the attester weight of a node in the registry.
+    /// @dev Only callable by the contract owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner whose attester weight will be changed.
+    /// @param _weight The new attester weight to assign to the node.
+    function changeAttesterWeight(address _nodeOwner, uint32 _weight) external onlyOwner {
+        _verifyNodeOwnerExists(_nodeOwner);
+        Node storage node = nodes[_nodeOwner];
+        if (_finalizeNodeRemoval(_nodeOwner, node)) {
+            return;
+        }
```

Comment:
I wonder if this can be bundled into a single getter function, something that returns `(node: Node, removed: bool)` I realise it only saves a single line, but just thought it's such a repeating pattern.

Reply:
How about this https://github.com/matter-labs/era-contracts/pull/555/commits/2c25a50a54564c4aa9676c9e2548864054b6659c ?